fineract-provider/src/main/java/org/apache/fineract/portfolio/address/service/AddressWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,279 @@
+
+		final Client client = this.clientRepository.getOne(clientId);
+		final boolean isActive = command.booleanPrimitiveValueOfParameterNamed("isActive");
+
+		final ClientAddress clientAddressobj = ClientAddress.fromJson(isActive, client, addobj, addressTypeIdObj);
+		this.clientAddressRepository.save(clientAddressobj);
+
+		return new CommandProcessingResultBuilder().withCommandId(command.commandId())
+				.withEntityId(clientAddressobj.getId()).build();
+	}
+
+	// following method is used for adding multiple addresses while creating new
+	// client
+
+	@Override
+	public CommandProcessingResult addNewClientAddress(final Client client, final JsonCommand command) {
+		CodeValue stateIdobj = null;
+		CodeValue countryIdObj = null;
+		long stateId;
+		long countryId;
+
+		final JsonArray addressArray = command.arrayOfParameterNamed("address");
+
+		for (int i = 0; i < addressArray.size(); i++) {
+			final JsonObject jsonObject = addressArray.get(i).getAsJsonObject();
+
+			// validate every address
+			this.fromApiJsonDeserializer.validateForCreate(jsonObject.toString(), true);
+
+			if (jsonObject.get("stateProvinceId") != null) {
+				stateId = jsonObject.get("stateProvinceId").getAsLong();
+				stateIdobj = this.codeValueRepository.getOne(stateId);
+			}
+
+			if (jsonObject.get("countryId") != null) {
+				countryId = jsonObject.get("countryId").getAsLong();
+				countryIdObj = this.codeValueRepository.getOne(countryId);
+			}
+
+			final long addressTypeId = jsonObject.get("addressTypeId").getAsLong();
+			final CodeValue addressTypeIdObj = this.codeValueRepository.getOne(addressTypeId);
+
+			final Address add = Address.fromJsonObject(jsonObject, stateIdobj, countryIdObj);
+			this.addressRepository.save(add);
+			final Long addressid = add.getId();
+			final Address addobj = this.addressRepository.getOne(addressid);
+
+			final boolean isActive = jsonObject.get("isActive").getAsBoolean();
+
+			final ClientAddress clientAddressobj = ClientAddress.fromJson(isActive, client, addobj, addressTypeIdObj);
+			this.clientAddressRepository.save(clientAddressobj);
+
+		}
+		final long typ = 1;
+		return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(typ).build();
+	}
+
+	@Override
+	public CommandProcessingResult updateClientAddress(final Long clientId, final JsonCommand command) {
+		this.context.authenticatedUser();
+
+		long stateId;
+
+		long countryId;
+
+		CodeValue stateIdobj;
+
+		CodeValue countryIdObj;
+
+		boolean is_address_update = false;
+
+		this.fromApiJsonDeserializer.validateForUpdate(command.json());
+
+		final long addressId = command.longValueOfParameterNamed("addressId");
+
+		final ClientAddress clientAddressObj = this.clientAddressRepositoryWrapper
+				.findOneByClientIdAndAddressId(clientId, addressId);
+
+		final Address addobj = this.addressRepository.getOne(addressId);
+
+		if (!(command.stringValueOfParameterNamed("street").isEmpty())) {
```

Comment:
If street is not passed in json, NullPointerException will thrown. This is same for all parameters in update json.